pynet/datasets/lpsnc.py

```diff
@@ -0,0 +1,206 @@
+"""
+
+# Imports
+from collections import namedtuple
+import os
+import logging
+import numpy as np
+import pandas as pd
+import nibabel as nib
+import progressbar
+from pynet.datasets import Fetchers
+
+# Global parameters
+MODALITIES = ('T2w', 'T1w', 'ce-GADOLINIUM_T1w', 'FLAIR')
+Item = namedtuple("Item", ["input_path", "output_path", "metadata_path"])
+logger = logging.getLogger("pynet")
+
+
+# datasetdir="/neurospin/radiomics/workspace_LPSNC/LPSNC_data"
+@Fetchers.register
+def fetch_lpsnc(datasetdir, modality=0):
+    """ Fetch/prepare the lpsnc dataset for pynet.
+
+    Parameters
+    ----------
+    datasetdir: str
+        the dataset destination folder.
+
+    Returns
+    -------
+    item: namedtuple
+        a named tuple containing 'input_path', 'output_path', and
+        'metadata_path'.
+    """
+    if modality not in MODALITIES and modality != 0:
+        raise ValueError(
+                "Expect modality==0 for all modalities "
+                + "or modality in ('T2w','T1w','ce-GADOLINIUM_T1w','FLAIR')")
+    logger.info("Loading lpsnc dataset.")
+
+    def _crop(arr):
+        # return arr[45: 195, 30: 220, 10: 145]
+        return arr
+
+    def _norm(arr):
+        arr = np.nan_to_num(arr)
+        logical_mask = (arr != 0)
+        mean = arr[logical_mask].mean()
+        std = arr[logical_mask].std()
+        return ((arr - mean) / std).astype(np.single)
+
+    def merge_rois(roifiles, v=1):
+        """ Merge the input ROIs.
+
+        Parameters
+        ----------
+        roifiles: list of str
+            the ROI images to be merged.
+        outdir: str
+            the destination folder.
+        fmaskname: str
+            the filename of the mask to be saved.
+
+        Returns
+        -------
+        merged_file: str
+            the merged ROI.
+        """
+        # Check inputs
+        if len(roifiles) == 0:
+            raise ValueError("Expect at least one ROI.")
+
+        # Merge ROIS
+        im = nib.load(roifiles[0])
+        ref_affine = im.affine
+        merged_data = np.nan_to_num(im.get_data())
+        if merged_data.ndim != 3:
+            raise ValueError("Expect 3d-ROIS.")
+        if len(roifiles) > 1:
+            for path in roifiles[1:]:
+                im = nib.load(path)
+                affine = im.affine
+                if not np.allclose(ref_affine, affine):
+                    raise ValueError(
+                            "ROI files have different affine matrices.")
+                data = np.nan_to_num(im.get_data())
+                if data.ndim != 3:
+                    raise ValueError("Expect 3d-ROIS.")
+                merged_data += data
+        merged_data[merged_data > 0] = v
+        return merged_data
+
+    mapping_path = os.path.join(datasetdir, "data.json")
+    zero_path = os.path.join(datasetdir, "zero.nii.gz")
+    # path for empty image of correct dimensions
+    zero = nib.load(zero_path).get_data()
```

Comment:
If this array is empty, you can just use np.zeros(shape, dtype=np.float) assuming the shape is known.